sources/packages/api/src/route-controllers/program-info-request/models/program-info-request.dto.ts

```diff
@@ -1,15 +1,22 @@
-import { IsNotEmpty, IsInt, Min, IsOptional } from "class-validator";
-import { SaveOfferingDTO } from "../../education-program-offering/models/education-program-offering.dto";
+import {
+  IsNotEmpty,
+  IsInt,
+  Min,
+  IsOptional,
+  IsPositive,
+} from "class-validator";
+import { SaveOfferingDTO } from "src/route-controllers/education-program-offering/models/education-program-offering.dto";
 import { ProgramInfoStatus } from "../../../database/entities";
 import { OfferingIntensity } from "../../../database/entities/offering-intensity.type";
 
-export interface CompleteProgramInfoRequestDto extends SaveOfferingDTO {
-  selectedProgram?: number;
-  selectedOffering?: number;
+export class CompleteProgramInfoRequestAPIInDTO {
+  @IsPositive()
+  selectedOffering: number;
```

Comment:
why was `selectedProgram` removed?

Reply:
It is not needed in the completeProgramInfoRequest method where the DTO is used as payload.

Reply:
okay.. do we need to check if the `selected offering id` belongs to the `selected program id` on the controller side?

Reply:
I believe that we can check if the offering belongs to the location, this would ensure the proper authorization.
IMO program is just a filter to have access to the offerings but since the API receives an offering id that belongs to the institution executing the PIR it should be ok.